Build failure: gencfsm

### Nixpkgs version

- Stable (25.05)

### Steps to reproduce

`nix build github:NixOS/nixpkgs?ref=nixos-25.05#gencfsm`

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
/nix/store/l0gm4vjnvm9kzczgglg6gwryiggl269x-vala-0.56.18/bin/valac --pkg=gmodule-2.0 --pkg=gtk+-3.0 --pkg=gio-2.0 --pkg=libsecret-1 --pkg=posix --pkg=encfs_wrapper --pkg=config --pkg=GEncfsM --vapidir=../src/vapi/ --basedir=../src/ --disable-warnings --pkg=gee-0.8    -C stash.vala main.vala add_stash.vala trayicon.vala dialogs.vala manager.vala preferences.vala mount_dialog.vala configure_stash.vala change_password.vala remove_stash.vala encfs/encfs_wrapper2.vala
encfs/encfs_wrapper.c: In function 'encfs_wrapper_is_mounted':
encfs/encfs_wrapper.c:43:32: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   43 |                 mount_dir_real = mount_dir;
      |                                ^
tools.c: In function 'client_quit_requested_cb':
tools.c:38:5: error: implicit declaration of function 'gnome_encfs_manager_on_logout' [-Wimplicit-function-declaration]
   38 |     gnome_encfs_manager_on_logout();
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:596: tools.o] Error 1
make[3]: *** Waiting for unfinished jobs....
mv -f gnome_encfs_manager_vala.stamp-t gnome_encfs_manager_vala.stamp
make[3]: Leaving directory '/build/gnome-encfs-manager-1.9/src'
make[2]: *** [Makefile:741: all-recursive] Error 1
make[2]: Leaving directory '/build/gnome-encfs-manager-1.9/src'
make[1]: *** [Makefile:452: all-recursive] Error 1
make[1]: Leaving directory '/build/gnome-encfs-manager-1.9'
make: *** [Makefile:384: all] Error 2
```

### Additional context

Compiles on 24.11. Downgrading to GCC 13 makes it compile again. I was too stupid to figure out how make gcc/valac/whoever ignore the warning 🙈

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.5, NixOS, 24.11 (Vicuna), 24.11.20250511.a39ed32`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.14`
 - nixpkgs: `/etc/nixpkgs/channels/nixpkgs`

### Notify maintainers

@spacefrogg

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ge/gencfsm/package.nix
+++ b/pkgs/by-name/ge/gencfsm/package.nix
@@ -29,6 +29,11 @@ stdenv.mkDerivation rec {
     sha256 = "RXVwg/xhfAQv3pWp3UylOhMKDh9ZACTuKM4lPrn1dk8=";
   };
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    # tools.c:38:5: error: implicit declaration of function 'gnome_encfs_manager_on_logout' []
+    "-Wno-implicit-function-declaration"
+  ];
+
   nativeBuildInputs = [
     autoconf
     automake

